dockersetup.py

```diff
@@ -0,0 +1,904 @@
+            raise argparse.ArgumentTypeError("The -u/--update and -r/--reinstall options are mutually exclusive")
+
+    socks_proxy_port = socks_proxy_host = ""
+    try:
+        if args.socks_proxy:
+            split = args.socks_proxy.split(":")
+            socks_proxy_port = split[2]
+            socks_proxy_host = split[1].replace("/", "")
+        elif args.http_proxy:
+            # Guess that this will work
+            split = args.http_proxy.split(":")
+            socks_proxy_port = '1080'
+            socks_proxy_host = split[1].replace("/", "")
+    except IndexError:
+        raise argparse.ArgumentTypeError("socks_proxy must be in format socks://<myproxy:port>")
+
+    if args.http_proxy and not args.https_proxy:
+        args.https_proxy = args.http_proxy
+    elif args.https_proxy and not args.http_proxy:
+        args.http_proxy = args.https_proxy
+
+    for entry in args.portmapping.split(','):
+        if len(entry.split(":")) != 2:
+            raise argparse.ArgumentTypeError("Port mapping must in the format HOST:CONTAINER. Ex: 8080:80. Multiple mappings should be separated by commas.")
+
+    if args.no_https:
+        if args.cert or args.cert_key or args.letsencrypt:
+            raise argparse.ArgumentTypeError("--no-https and --cert/--cert-key/--letsencrypt options are mutually exclusive")
+    if args.letsencrypt:
+        if args.cert or args.cert_key:
+            raise argparse.ArgumentTypeError("--letsencrypt and --cert/--cert-key options are mutually exclusive")
+    if args.cert and not os.path.exists(args.cert):
+        raise argparse.ArgumentTypeError("Specified certificate file %s does not exist" % args.cert)
+    if args.cert_key and not os.path.exists(args.cert_key):
+        raise argparse.ArgumentTypeError("Specified certificate key file %s does not exist" % args.cert_key)
+    if args.cert_key and not args.cert:
+        raise argparse.ArgumentTypeError("Certificate key file specified but not certificate")
+    if args.cert and not args.cert_key:
+        args.cert_key = os.path.splitext(args.cert)[0] + '.key'
+        if not os.path.exists(args.cert_key):
+            raise argparse.ArgumentTypeError("Could not find certificate key, please use --cert-key to specify it")
+
+    email_host = None
+    email_port = None
+    if args.email_host:
+        email_host_split = args.email_host.split(':')
+        email_host = email_host_split[0]
+        if len(email_host_split) > 1:
+            email_port = email_host_split[1]
+
+    if args.email_ssl and args.email_tls:
+        raise argparse.ArgumentTypeError("--email-ssl and --email-tls options are mutually exclusive")
+    if (args.email_ssl or args.email_tls or args.email_user or args.email_password) and not email_host:
+        raise argparse.ArgumentTypeError("If any of the email host options are specified then you must also specify an email host with -e/--email-host")
+
+    return args, socks_proxy_port, socks_proxy_host, email_host, email_port
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are unpacking more than three variables while using a function that returns more than one variable. Using a large number of return values is prone to errors. We recommend that you return a small class or `namedtuple` instance instead.

[Learn more](https://docs.python.org/3/library/typing.html#typing.NamedTuple)